### PR TITLE
Add option to build shared library with cmake

### PR DESCRIPTION
**Pull request by robert-kalmar**
The shared library build correspond to the build rules in bazel for shared library.

### CODE REVIEW

File: tensorflow/lite/CMakeLists.txt [hunk 1]
```diff
@@ -623,14 +642,27 @@ set(_ALL_TFLITE_SRCS
   ${TFLITE_SOURCE_DIR}/schema/schema_utils.cc
   ${TFLITE_SOURCE_DIR}/schema/schema_generated.h
 )
-add_library(tensorflow-lite
-  ${_ALL_TFLITE_SRCS}
```
Comment (qukhan) [inline tensorflow/lite/CMakeLists.txt:627]: If `TFLITE_BUILD_SHARED_LIB` is false, where is the `add_library` call?

File: tensorflow/lite/CMakeLists.txt [hunk 2]
```diff
@@ -623,14 +642,27 @@ set(_ALL_TFLITE_SRCS
...
+  else ()
+    target_link_options(tensorflow-lite PRIVATE "-Wl,-z,defs")
+    target_link_options(tensorflow-lite PRIVATE "-Wl,--version-script,${TENSORFLOW_SOURCE_DIR}/tensorflow/lite/tflite_version_script.lds")
+  endif()
+endif()
+
+set_target_properties(tensorflow-lite
+  PROPERTIES
+    VERSION ${PROJECT_VERSION}
+    SOVERSION  ${PROJECT_VERSION}
```
Comment (qukhan) [inline tensorflow/lite/CMakeLists.txt:659]: How does this behave for static libraries?

Comment (robert-kalmar) [reply tensorflow/lite/CMakeLists.txt:659]: CMake applies the `VERSION` and `SOVERSION` properties only for shared library - https://cmake.org/cmake/help/latest/prop_tgt/VERSION.html , https://cmake.org/cmake/help/latest/prop_tgt/SOVERSION.html .

File: tensorflow/lite/CMakeLists.txt (hunk 1 above)
Comment (robert-kalmar) [reply tensorflow/lite/CMakeLists.txt:627]: It is above, here https://github.com/tensorflow/tensorflow/pull/62933/files#diff-f22833f37d31fb773c988ab1a12b7c6a3265b1ee5eb3fe93c885a02a9f1d3d20R604, but I oversight the patch removes the `_ALL_TFLITE_SRCS` variable. Let us fix it.

Comment (robert-kalmar) [reply tensorflow/lite/CMakeLists.txt:627]: Hi @qukhan, we fixed the patch to correspond with the latest changes. Please re-review it. 
Thanks,

Comment (robert-kalmar) [reply tensorflow/lite/CMakeLists.txt:627]: @qukhan , we fixed the PR to corresponds to the latest changes. Please review it. 
Thanks,

**Review by qukhan — APPROVED** [review]

**Comment by gbaned** [comment]
Hi @robert-kalmar Can you please rebase your branch and resolve the conflicts? Thank you!

**Comment by keerthanakadiri** [comment]
Hi @robert-kalmar Can you please resolve the conflicts? Thank you!

**Comment by keerthanakadiri** [comment]
Hi @robert-kalmar Can you please resolve the conflicts? Thank you!

**Comment by keerthanakadiri** [comment]
Hi @robert-kalmar Can you please resolve the conflicts? Thank you!

**Comment by keerthanakadiri** [comment]
Hi @robert-kalmar, Can you please resolve the conflicts? Thank you!

**Pull request closed without merging**